yt/fields/tests/test_magnetic_fields.py

```diff
@@ -19,19 +23,23 @@ def test_magnetic_fields():
 
     ds1 = load_uniform_grid(data1, ddims, unit_system="cgs")
     ds2 = load_uniform_grid(data2, ddims, unit_system="mks")
+    ds3 = yt.load(gizmo_mhd, bounding_box=[[-400, 400]]*3, unit_system="code")
 
     ds1.index
     ds2.index
+    ds3.index
 
     dd1 = ds1.all_data()
     dd2 = ds2.all_data()
+    dd3 = ds3.all_data()
 
     assert ds1.fields.gas.magnetic_field_strength.units == "gauss"
     assert ds1.fields.gas.magnetic_field_poloidal.units == "gauss"
     assert ds1.fields.gas.magnetic_field_toroidal.units == "gauss"
     assert ds2.fields.gas.magnetic_field_strength.units == "T"
     assert ds2.fields.gas.magnetic_field_poloidal.units == "T"
     assert ds2.fields.gas.magnetic_field_toroidal.units == "T"
+    assert ds3.fields.PartType0.particle_magnetic_field.units == "code_magnetic"
```

Comment:
I'm not sure if this test is enough. The tests below cannot be performed on ds3 because I haven't added all the derived magnetic fields to the Gizmo frontend.

Reply:
I'd test to make sure you can access these fields and that e.g. the vector fields and component fields, as well as the aliases to internal yt field names are the correct shapes and have the correct units.

With the demeshening you'd be able to use the existing definitions for derived magnetic fields. Unfortunately I think to get those working here you'll need to define particle-specific versions of them.

Also I'd prefer it if this test were in the Gizmo frontend, even if it duplicates some boilerplate from the existing test you've modified here. The way you've modified this existing test means that the test will only run if the GIZMO MHD dataset is present in the dataset search path. That means that this test will no longer run on travis, for example.

Reply:
Thanks, I'll change them as you suggested.